apps/vaccitron/src/app/report-list-page/report-list-page.component.ts

```diff
@@ -72,8 +86,38 @@ export class ReportListPageComponent implements OnInit, OnDestroy {
     this.#onDestroy$.next();
   }
 
-  reportsListActionEvent($event) {
+  reportsListActionEvent($event: ReportsListAction) {
     if (!this.isLoggedIn) return this.openLoginDialog();
+    switch ($event.type) {
+      case 'comingFeedback':
+        const dialogRef = this.dialog.open(MatSpinner, {
+          direction: 'rtl',
+          autoFocus: false,
+        });
+
+        this.vaccinesReportsService
+          .updateImComing(environment.apiUrl + environment.comingFeedback, {
```

Comment:
I'm not sure it's a good idea to have a blocking modal in this case. Some form of nonblocking spinner is better (unless you have some concerns regards parallel user operations that you prefer not to deal?).

Reply:
Look now. Hacked it so it will make the button `diabled` while waiting for a response.